### PR TITLE
add svg support

### PR DESCRIPTION
**Pull request by MarinaPasternak**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -4,11 +4,16 @@ import { Provider } from 'react-redux';
...
-    <Counter />
-  </Provider>
+  <>
+    <Provider store={store}>
+      <h1>Hello, I'm App Component!</h1>
+      <Counter />
+    </Provider>
+    <img src={logo} alt="" />
+  </> 
 );
```
Comment (PavelKundenko) [inline src/app.js:19]: Better place image inside `Provider` and remove fragment.
```
export const App = () => (
 <Provider store={store}>
    <h1>Hello, I'm App Component!</h1>
    <Counter />
    <img src={logo} alt="" />	     
  </Provider>
);
```

Comment (sarhan-azizov) [reply src/app.js:19]: I recommend creating an Icon component as an example. However, the `img` tag is bad practice. ref: https://medium.com/@gineetmehta/how-to-use-svg-sprites-regular-html-vs-react-9f98948c03f0

Also, I would advise you to investigate this topic. Some articles you can find below 
https://habr.com/ru/post/327700/
https://developer.mozilla.org/en-US/docs/Web/SVG/Element/use#Browser_compatibility
https://github.com/JetBrains/svg-sprite-loader/blob/master/2.0.md

The example in the Icon component you can find below

```JSX
import React, { PureComponent } from 'react';
import { string, bool, oneOf, func } from 'prop-types';
import classNames from 'classnames';
import { values } from 'lodash';

import { withRender } from '../shared';
import { SIZES } from '../constants';
import styles from './component.scss';

export class Icon extends PureComponent {
    render() {
        const iconProps = {
            className: classNames(styles.icon, this.props.className, {
                [styles[this.props.size]]: this.props.size
            }),
            onClick: this.props.disabled ? undefined : this.props.onClick,
            tabIndex: this.props.onClick && !this.props.disabled ? 0 : undefined,
            role: this.props.onClick && !this.props.disabled ? 'button' : undefined,
            ref: this.props.innerRef
        };
        const glyphClassName = classNames(styles.glyph, this.props.glyphClassName, {
            [styles.clickable]: this.props.onClick,
            [styles.controlled]: this.props.controlled,
            [styles.active]: this.props.active,
            [styles.disabled]: this.props.disabled,
            [styles[this.props.size]]: this.props.size
        });

        // the key prop in svg need for fixing the IE issue related to external content in SVG
        // https://stackoverflow.com/questions/22516712/svg-use-tag-with-external-reference-in-ie-11
        return (
            <i {...iconProps}>
                <svg key={this.props.glyph} className={glyphClassName}>
                    <use xlinkHref={`#${this.props.glyph}`} />
                </svg>
            </i>
        );
    }
}

Icon.propTypes = {
    glyph: string.isRequired,
    size: oneOf(values(SIZES)),

    controlled: bool,
    active: bool,
    disabled: bool,

    className: string,
    glyphClassName: string,

    innerRef: func,
    onClick: func
};

Icon.defaultProps = {
    size: SIZES.MEDIUM,

    controlled: true,
    active: false,
    disabled: false,

    className: undefined,
    glyphClassName: undefined,

    innerRef: undefined,
    onClick: undefined
};

export default Icon;
```

```SCSS
// icon.scss
@import '../scss/theme';

.icon {
  display: inline-flex;
}

.glyph {
  fill: currentColor;

  &:hover:not(.controlled) {
    fill: $c-french-blue;
  }
}

.clickable {
  cursor: pointer;
}

.active {
  fill: $c-french-blue;
}

.disabled {
  color: $c-pastel-gray;
}

.mini {
  width: $gs-mini;
  height: $gs-mini;
}

.tiny {
  width: $gs-tiny;
  height: $gs-tiny;
}

.small {
  width: $gs-small;
  height: $gs-small;
}

.medium {
  width: $gs-medium;
  height: $gs-medium;
}

.large {
  width: $gs-large;
  height: $gs-large;
}

.big {
  width: $gs-big;
  height: $gs-big;
}

.huge {
  width: $gs-huge;
  height: $gs-huge;
}

.massive {
  width: $gs-massive;
  height: $gs-massive;
}

.banner {
  width: $gs-banner;
  height: $gs-banner;
}

// _sizes.scss
// Glyph sizes
$gs-mini: 10px;
$gs-tiny: 12px;
$gs-small: 14px;
$gs-medium: 16px;
$gs-large: 18px;
$gs-big: 20px;
$gs-huge: 22px;
$gs-massive: 24px;
$gs-banner: 48px;

// Border radius
$b-radius: 3px;
```

File: webpack.config.js [hunk 2]
```diff
@@ -15,6 +15,19 @@ module.exports = {
         exclude: /node_modules/,
         use: ['babel-loader'],
       },
+      {
```
Comment (sarhan-azizov) [inline webpack.config.js:18]:
```javascript
{
                test: /\.svg$/,
                include: path.resolve(__dirname, '../../src/components/icon/svg'),
                use: [
                    {
                        loader: 'svg-sprite-loader'
                    },
                    {
                        loader: 'svgo-loader',
                        options: {
                            plugins: [
                                { removeTitle: true },
                                { convertColors: { shorthex: true } },
                                { removeStyleElement: true }
                            ]
                        }
                    }
                ]
            },
```

**Review by sarhan-azizov — CHANGES_REQUESTED** [review]

**Review by archontis-cpu — CHANGES_REQUESTED** [review]
Please, be attentive with `package-lock.json` | `yarn.lock` when installing or removing any dependencies by hand (like "open `package.json` and delete a line with an undesirable package at `dependencies` or `devDependencies`"). This may cause conflicts with the dependency graph, git, bundling, etc.
So, when you are removing a package by hand, you need to update `package-lock.json`, and, probably, `node_modules`:
1. Delete `node_modules` folder.
2. Delete `package-lock.json`.
3. Open a terminal and run `npm install`. This command always generates `package-lock.json`.
4. Commit and push changes.

But for the future, use `npm install | uninstall <package-name>` to avoid such problems.

File: src/icon.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+export default function Icon({glyph, viewBox = '0 0 16 16', className = 'icon',...props}){
```
Comment (sarhan-azizov) [inline src/icon.js:1]: Why Didn't you use the proposed interface by me?

File: src/app.js [hunk 4]
```diff
@@ -1,14 +1,20 @@
...
 
 export const App = () => (
-  <Provider store={store}>
-    <h1>Hello, I'm App Component!</h1>
-    <Counter />
-  </Provider>
+    <Provider store={store}>
+      <h1>Hello, I'm App Component!</h1>
+      <Counter />
+      <div>
```
Comment (sarhan-azizov) [inline src/app.js:16]:
```suggestion
      <Icon glyph="cat" color="red" size="big" />
```

**Review by sarhan-azizov — CHANGES_REQUESTED** [review]

File: src/icon.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+    return (
+      <svg className={className} viewBox={viewBox} {...props}>
+        <use xlinkHref={`#${glyph}`} />
```
Comment (sarhan-azizov) [inline src/icon.js:4]:
```suggestion
        <use xlinkHref={`#${require('./svg/${props.glyph}.svg')}`} />
```

**Review by sarhan-azizov — CHANGES_REQUESTED** [review]

**Comment by MarinaPasternak** [comment]
image

**Review by sarhan-azizov — APPROVED** [review]

**Review by MozdolevskyiMaksym — APPROVED** [review]

**Review by Anastasiia0999 — APPROVED** [review]

**Review by PavelKundenko — APPROVED** [review]

**Review by archontis-cpu — APPROVED** [review]

**Pull request merged**